synfig-studio/src/gui/trees/historytreestore.cpp

```diff
@@ -209,6 +212,7 @@ HistoryTreeStore::on_redo_stack_cleared()
 void
 HistoryTreeStore::on_new_action(etl::handle<synfigapp::Action::Undoable> action)
 {
+ if (!synfigapp::Action::System::block_new_history){ /*if change isn't due to rapid repeated actions show history normally*/
```

Comment:
Is this part is still needed?

Reply:
seems as if it is redundant after the new implementation (because the new action isn't made when history is blocked so this function shouldn't be called when history is blocked), I also tested it to make sure it is.